modules/images/image-loading-optimization/optimization.php

```diff
@@ -180,44 +185,71 @@ function ilo_optimize_template_output_buffer( string $buffer ): string {
 	// Walk over all IMG tags in the document and ensure fetchpriority is set/removed, and gather IMG attributes for preloading.
 	$processor = new ILO_HTML_Tag_Processor( $buffer );
 	foreach ( $processor->open_tags() as $tag_name ) {
-		if ( 'IMG' !== $tag_name ) {
+		$is_img_tag = ( 'IMG' === $tag_name );
+		$style      = $processor->get_attribute( 'style' );
+
+		/*
+		 * Note that CSS allows for a `background`/`background-image` to have multiple `url()` CSS functions, resulting
+		 * in multiple background images being layered on top of each other. This ability is not employed in core. Here
+		 * is a regex to search WPDirectory for instances of this: /background(-image)?:[^;}]+?url\([^;}]+?[^_]url\(/.
+		 * It is used in Jetpack with the second background image being a gradient. To support multiple background
+		 * images, this logic would need to be modified to make $background_image an array and to have a more robust
+		 * parser of the `url()` functions from the property value.
+		 */
+		$background_image = null;
+		if ( $style && preg_match( '/background(-image)?\s*:[^;]*?url\(\s*[\'"]?(?!data:)(?<background_image>.+?)[\'"]?\s*\)/', $style, $matches ) ) {
+			$background_image = $matches['background_image'];
+		}
+
+		if ( ! ( $is_img_tag || $background_image ) ) {
 			continue;
 		}
 
 		$xpath = $processor->get_xpath();
 
 		// Ensure the fetchpriority attribute is set on the element properly.
-		if ( $common_lcp_element && $xpath === $common_lcp_element['xpath'] ) {
-			if ( 'high' === $processor->get_attribute( 'fetchpriority' ) ) {
-				$processor->set_attribute( 'data-ilo-fetchpriority-already-added', true );
-			} else {
-				$processor->set_attribute( 'fetchpriority', 'high' );
-				$processor->set_attribute( 'data-ilo-added-fetchpriority', true );
-			}
+		if ( $is_img_tag ) {
+			if ( $common_lcp_element && $xpath === $common_lcp_element['xpath'] ) {
+				if ( 'high' === $processor->get_attribute( 'fetchpriority' ) ) {
+					$processor->set_attribute( 'data-ilo-fetchpriority-already-added', true );
+				} else {
+					$processor->set_attribute( 'fetchpriority', 'high' );
+					$processor->set_attribute( 'data-ilo-added-fetchpriority', true );
+				}
 
-			// Never include loading=lazy on the LCP image common across all breakpoints.
-			if ( 'lazy' === $processor->get_attribute( 'loading' ) ) {
-				$processor->set_attribute( 'data-ilo-removed-loading', $processor->get_attribute( 'loading' ) );
-				$processor->remove_attribute( 'loading' );
+				// Never include loading=lazy on the LCP image common across all breakpoints.
+				if ( 'lazy' === $processor->get_attribute( 'loading' ) ) {
+					$processor->set_attribute( 'data-ilo-removed-loading', $processor->get_attribute( 'loading' ) );
+					$processor->remove_attribute( 'loading' );
+				}
+			} elseif ( $all_breakpoints_have_url_metrics && $processor->get_attribute( 'fetchpriority' ) ) {
```

Comment:
this bit is unclear to me. if all breakpoints have data, you are falling back to the server heuristics?

Reply:
If all breakpoints have data, then we can reliably use the `preload` links to prioritize loading of the LCP image. In that case, we can remove the `fetchpriority` attribute from the LCP image here.

Reply:
I see, so you are favoring preload over fetchpriority? maybe we discussed this before, couldn't you use both of them? or is high priority already implied when using preload?

Reply:
The `preload` links also include `fetchpriority=high`, so having it on the element is redundant. Also, it can be counter productive to be on the `img` in the case where there are different LCP images for different viewports. If there is an image with `fetchpriority=high` but it is only for the desktop viewport then it will get loaded with high priority even for mobile users. The preload links also include `media` attribute with media queries so they enable `fetchpriority=high` to be conditional on the viewport size.